controllers/humiocluster_status.go

```diff
@@ -56,6 +56,15 @@ func (r *HumioClusterReconciler) setVersion(ctx context.Context, version string,
 	}
 }
 
+func (r *HumioClusterReconciler) setLicense(ctx context.Context, licenseStatus humiov1alpha1.HumioLicenseStatus, hc *humiov1alpha1.HumioCluster) {
```

Comment:
I'm starting to think we should rename the functions inside this file. Right now it is not that clear what `setLicense` actually means.

To me, the name indicates that it sets the license in the Humio cluster, similar to if the function was called `activateLicense` or perhaps `installLicense`.

This is probably a more general thing about all the functions we have that sets/updates the `HumioCluster.Status` fields, so I'm also good with keeping it the way it is for now, but I do think we should consider doing something to make it more clear when reading the code that these functions are "only" updating the object status.

Reply:
Yes I agree, and a refactor in general should be done to make a better pattern for the status updates.